File: app/src/main/java/bible/translationtools/converter/TransformerTask.java
package bible.translationtools.converter;

import android.os.AsyncTask;
import android.support.v4.app.Fragment;

/**
 * Created by mXaln on 11/11/17.
 */

public final class TransformerTask extends AsyncTask<Void, Integer, Integer> {

    TransformerResultCallback callback;

    public TransformerTask(Fragment c) {
        this.callback = (TransformerResultCallback) c;
    }

    @Override
    protected Integer doInBackground(Void... params) {
        return callback.startTransformation();
    }

    @Override
    protected void onPreExecute() {
        callback.transformationStarted();
        super.onPreExecute();
    }

    @Override
    protected void onProgressUpdate(Integer... values) {
        super.onProgressUpdate(values);
    }

    @Override
    protected void onPostExecute(Integer result) {
        callback.transformationDone(result);
        super.onPostExecute(result);
    }

    public interface TransformerResultCallback {
        Integer startTransformation();
        Void transformationStarted();
        Void transformationDone(Integer result);
    }
}

File: app/src/main/java/bible/translationtools/converter/TransformerTask.kt
package bible.translationtools.converter

import android.os.AsyncTask
import androidx.fragment.app.Fragment

/**
 * Created by mXaln on 11/11/17.
 */
class TransformerTask(c: Fragment?) : AsyncTask<Void?, Int?, Int?>() {
    var callback: TransformerResultCallback

    init {
        this.callback = c as TransformerResultCallback
    }

    override fun doInBackground(vararg params: Void?): Int? {
        return callback.startTransformation()
    }

    override fun onPreExecute() {
        callback.transformationStarted()
        super.onPreExecute()
    }

    override fun onProgressUpdate(vararg values: Int?) {
        super.onProgressUpdate(*values)
    }

    override fun onPostExecute(result: Int?) {
        callback.transformationDone(result)
        super.onPostExecute(result)
    }

    interface TransformerResultCallback {
        fun startTransformation(): Int
        fun transformationStarted()
        fun transformationDone(result: Int?)
    }
}
